client/main.lua

```diff
@@ -199,40 +246,79 @@ CreateThread(function()
-					DrawText3D(Config.Locations["freedom"].coords.x, Config.Locations["freedom"].coords.y, Config.Locations["freedom"].coords.z, "Check time")
-				end
+freedom = BoxZone:Create(vector3(Config.Locations["freedom"].coords.x, Config.Locations["freedom"].coords.y, Config.Locations["freedom"].coords.z), 2.75, 2.75, {
+	name="freedom",
+	debugPoly = false,
+})
+freedom:onPlayerInOut(function(isPointInside)
+	if isPointInside then
+		exports['qb-drawtext']:DrawText('[E] Check Time', 'left')
+		insidefreedom = true
+	else
+		insidefreedom = false
+		exports['qb-drawtext']:HideText()
+	end
+end)
 
-				if #(pos - vector3(Config.Locations["shop"].coords.x, Config.Locations["shop"].coords.y, Config.Locations["shop"].coords.z)) < 1.5 then
-					DrawText3D(Config.Locations["shop"].coords.x, Config.Locations["shop"].coords.y, Config.Locations["shop"].coords.z, "~g~E~w~ - Canteen")
-					if IsControlJustReleased(0, 38) then
-                        local ShopItems = {}
-                        ShopItems.label = "Prison Canteen"
-                        ShopItems.items = Config.CanteenItems
-                        ShopItems.slots = #Config.CanteenItems
-                        TriggerServerEvent("inventory:server:OpenInventory", "shop", "Canteenshop_"..math.random(1, 99), ShopItems)
-					end
-					DrawMarker(2, Config.Locations["shop"].coords.x, Config.Locations["shop"].coords.y, Config.Locations["shop"].coords.z, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.3, 0.3, 0.2, 255, 55, 22, 222, false, false, false, 1, false, false, false)
-				elseif #(pos - vector3(Config.Locations["shop"].coords.x, Config.Locations["shop"].coords.y, Config.Locations["shop"].coords.z)) < 2.5 then
-					DrawText3D(Config.Locations["shop"].coords.x, Config.Locations["shop"].coords.y, Config.Locations["shop"].coords.z, "Canteen")
-					DrawMarker(2, Config.Locations["shop"].coords.x, Config.Locations["shop"].coords.y, Config.Locations["shop"].coords.z, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.3, 0.3, 0.2, 255, 55, 22, 222, false, false, false, 1, false, false, false)
-				elseif #(pos - vector3(Config.Locations["shop"].coords.x, Config.Locations["shop"].coords.y, Config.Locations["shop"].coords.z)) < 10 then
-					DrawMarker(2, Config.Locations["shop"].coords.x, Config.Locations["shop"].coords.y, Config.Locations["shop"].coords.z, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.3, 0.3, 0.2, 255, 55, 22, 222, false, false, false, 1, false, false, false)
-				end
+canteen = BoxZone:Create(vector3(Config.Locations["shop"].coords.x, Config.Locations["shop"].coords.y, Config.Locations["shop"].coords.z), 2.75, 7.75, {
+name="canteen",
+debugPoly = false,
+})
+canteen:onPlayerInOut(function(isPointInside)
+	if isPointInside then
+		exports['qb-drawtext']:DrawText('[E] Open Canteen', 'left')
+		insidecanteen = true
+	else
+		insidecanteen = false
+		exports['qb-drawtext']:HideText()
+	end
+end)
+
+local function interaction()
+	if Config.UseTarget then
+			print(freedom_ped, canteen_ped)
```

Comment:
Leftover print

Reply:
Resolving. Added this export before drawtext was merged with core